Build failure: riscv-pk

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Either attempt to build `pkgs.spike` or `pkgs.pkgsCross.riscv64-embedded.riscv-pk`.

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by RISC-V Proxy Kernel configure 1.0.0, which was
generated by GNU Autoconf 2.72.  Invocation command line was

  $ ../configure --prefix=/nix/store/21cgj30hlk5ys6ax0vzfj64fpy5nrkqp-riscv-pk-riscv64-none-elf-1.0.0-unstable-2024-10-09 --build=x86_64-unknown-linux-gnu --host=riscv64-none-elf

## --------- ##
## Platform. ##
## --------- ##

hostname = localhost
uname -m = x86_64
uname -r = 6.12.49
uname -s = Linux
uname -v = #1-NixOS SMP PREEMPT_DYNAMIC Thu Sep 25 09:13:51 UTC 2025

/usr/bin/uname -p = unknown
/bin/uname -X     = unknown

/bin/arch              = unknown
/usr/bin/arch -k       = unknown
/usr/convex/getsysinfo = unknown
/usr/bin/hostinfo      = unknown
/bin/machine           = unknown
/usr/bin/oslevel       = unknown
/bin/universe          = unknown

PATH: /nix/store/vgsiyafmzsrrdsk1z76sqjc5xhds5n74-autoconf-2.72/bin/
PATH: /nix/store/r4zybqyg78gk4vrq3hffwchr2vpjj369-automake-1.18.1/bin/
PATH: /nix/store/vxhqq9paalnry3qaxlfhiqda9886akd3-gettext-0.25.1/bin/
PATH: /nix/store/vhvpmlp8v9kjy5rk0halj1basab6nk34-libtool-2.5.4/bin/
PATH: /nix/store/zsi23zdblfiazn37xv139yr0i9xvb8rb-gnum4-1.4.20/bin/
PATH: /nix/store/zb630w748n812dyr2nx93xpmn06aks53-file-5.45/bin/
PATH: /nix/store/0032synbfqd0ylgf9szgp724nmw2fq89-patchelf-0.15.2/bin/
PATH: /nix/store/w8cdh38i58n7073d4gcfbhbfzilbp7ky-riscv64-none-elf-gcc-wrapper-14.3.0/bin/
PATH: /nix/store/k0ibrfk0y4wd02y231y6njzhyskr3xqa-riscv64-none-elf-gcc-14.3.0/bin/
PATH: /nix/store/00bc157nm93q5fjz551fwk60ihlbilvj-coreutils-9.7/bin/
PATH: /nix/store/j6nqhfpmf2fpfnya5gmzkaz34ln0vg25-riscv64-none-elf-binutils-wrapper-2.44/bin/
PATH: /nix/store/yqkzfrjan602y32zz0k4wa6zl18xdcg9-riscv64-none-elf-binutils-2.44/bin/
PATH: /nix/store/00bc157nm93q5fjz551fwk60ihlbilvj-coreutils-9.7/bin/
PATH: /nix/store/wq1cxh4s8rz8vs4ssp2z49v96m0nhxhj-findutils-4.10.0/bin/
PATH: /nix/store/c7qlkpbivy2znqxd25prgh9pwjxdri6w-diffutils-3.12/bin/
PATH: /nix/store/7zy6jnnpqh16b18j6ar289vhwp17msh0-gnused-4.9/bin/
PATH: /nix/store/6q57mnfak3q8ss656qgp245qzqvnniyr-gnugrep-3.12/bin/
PATH: /nix/store/swbmnc2s8g2zxzxh62hx1vbbcg6i2r5b-gawk-5.3.2/bin/
PATH: /nix/store/n09s57l86zxyjydkxagawxpkhh3d3dlh-gnutar-1.35/bin/
PATH: /nix/store/wd4i398mh329vhw93mvmamw3hrnhl257-gzip-1.14/bin/
PATH: /nix/store/xnd0ld68fzj1g1qr19gg0va486snwj22-bzip2-1.0.8-bin/bin/
PATH: /nix/store/lmn1m0giq5bkgdqvdpiryb9a2ly28w6g-gnumake-4.4.1/bin/
PATH: /nix/store/cl2gkgnh26mmpka81pc2g5bzjfrili92-bash-5.3p3/bin/
PATH: /nix/store/r9lif1a0naanr8kdapwnbnr39lr7d56a-patch-2.8/bin/
PATH: /nix/store/77jyjqy2b9l2a0139im9ds28zkdv0ybc-xz-5.8.1-bin/bin/
PATH: /nix/store/5qm0vivgc7lmiyfbf5d5a55s40d996bd-file-5.45/bin/


## ----------- ##
## Core tests. ##
## ----------- ##

configure:2518: looking for aux files: install-sh config.guess config.sub
configure:2531:  trying ../scripts/
configure:2542:   ../scripts/install-sh found
configure:2560:   ../scripts/config.guess found
configure:2560:   ../scripts/config.sub found
configure:2681: checking build system type
configure:2697: result: x86_64-unknown-linux-gnu
configure:2717: checking host system type
configure:2732: result: riscv64-none-elf
configure:2774: checking for riscv64-none-elf-gcc
configure:2807: result: riscv64-none-elf-gcc
configure:3212: checking for C compiler version
configure:3221: riscv64-none-elf-gcc --version >&5
riscv64-none-elf-gcc (GCC) 14.3.0
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

configure:3232: $? = 0
configure:3221: riscv64-none-elf-gcc -v >&5
Using built-in specs.
COLLECT_GCC=/nix/store/k0ibrfk0y4wd02y231y6njzhyskr3xqa-riscv64-none-elf-gcc-14.3.0/bin/riscv64-none-elf-gcc
COLLECT_LTO_WRAPPER=/nix/store/k0ibrfk0y4wd02y231y6njzhyskr3xqa-riscv64-none-elf-gcc-14.3.0/libexec/gcc/riscv64-none-elf/14.3.0/lto-wrapper
Target: riscv64-none-elf
Configured with: ../gcc-14.3.0/configure --prefix=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-riscv64-none-elf-gcc-14.3.0 --with-gmp-include=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gmp-with-cxx-6.3.0-dev/include --with-gmp-lib=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gmp-with-cxx-6.3.0/lib --with-mpfr-include=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-mpfr-4.2.2-dev/include --with-mpfr-lib=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-mpfr-4.2.2/lib --with-mpc=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libmpc-1.3.1 --with-native-system-header-dir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-newlib-riscv64-none-elf-4.5.0.20241231/riscv64-none-elf/include --with-build-sysroot=/ --with-gxx-include-dir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-riscv64-none-elf-gcc-14.3.0/include/c++/14.3.0/ --program-prefix=riscv64-none-elf- --enable-lto --disable-libstdcxx-pch --without-included-gettext --with-system-zlib --enable-static --enable-languages=c,c++ --disable-multilib --disable-shared --enable-plugin --with-isl=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-isl-0.20 --with-as=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-riscv64-none-elf-binutils-wrapper-2.44/bin/riscv64-none-elf-as --with-headers=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-newlib-riscv64-none-elf-4.5.0.20241231/riscv64-none-elf/include --enable-__cxa_atexit --enable-long-long --enable-threads=single --enable-nls --with-newlib --disable-bootstrap --enable-default-pie --build=x86_64-unknown-linux-gnu --host=x86_64-unknown-linux-gnu --target=riscv64-none-elf
Thread model: single
Supported LTO compression algorithms: zlib
gcc version 14.3.0 (GCC) 
configure:3232: $? = 0
configure:3221: riscv64-none-elf-gcc -V >&5
riscv64-none-elf-gcc: error: unrecognized command-line option '-V'
riscv64-none-elf-gcc: fatal error: no input files
compilation terminated.
configure:3232: $? = 1
configure:3221: riscv64-none-elf-gcc -qversion >&5
riscv64-none-elf-gcc: error: unrecognized command-line option '-qversion'; did you mean '--version'?
riscv64-none-elf-gcc: fatal error: no input files
compilation terminated.
configure:3232: $? = 1
configure:3221: riscv64-none-elf-gcc -version >&5
riscv64-none-elf-gcc: error: unrecognized command-line option '-version'
riscv64-none-elf-gcc: fatal error: no input files
compilation terminated.
configure:3232: $? = 1
configure:3252: checking whether the C compiler works
configure:3274: riscv64-none-elf-gcc    conftest.c  >&5
/nix/store/yqkzfrjan602y32zz0k4wa6zl18xdcg9-riscv64-none-elf-binutils-2.44/bin/riscv64-none-elf-ld: -pie not supported
collect2: error: ld returned 1 exit status
configure:3278: $? = 1
configure:3319: result: no
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "RISC-V Proxy Kernel"
| #define PACKAGE_TARNAME "riscv-pk"
| #define PACKAGE_VERSION "1.0.0"
| #define PACKAGE_STRING "RISC-V Proxy Kernel 1.0.0"
| #define PACKAGE_BUGREPORT "Andrew Waterman"
| #define PACKAGE_URL ""
| /* end confdefs.h.  */
| 
| int
| main (void)
| {
| 
|   ;
|   return 0;
| }
configure:3324: error: in '/build/source/build':
configure:3326: error: C compiler cannot create executables
See 'config.log' for more details

## ---------------- ##
## Cache variables. ##
## ---------------- ##

ac_cv_build=x86_64-unknown-linux-gnu
ac_cv_env_CCC_set=
ac_cv_env_CCC_value=
ac_cv_env_CC_set=set
ac_cv_env_CC_value=riscv64-none-elf-gcc
ac_cv_env_CFLAGS_set=
ac_cv_env_CFLAGS_value=
ac_cv_env_CPPFLAGS_set=
ac_cv_env_CPPFLAGS_value=
ac_cv_env_CXXFLAGS_set=
ac_cv_env_CXXFLAGS_value=
ac_cv_env_CXX_set=set
ac_cv_env_CXX_value=riscv64-none-elf-g++
ac_cv_env_LDFLAGS_set=
ac_cv_env_LDFLAGS_value=
ac_cv_env_LIBS_set=
ac_cv_env_LIBS_value=
ac_cv_env_RISCV_set=
ac_cv_env_RISCV_value=
ac_cv_env_STOW_PREFIX_set=
ac_cv_env_STOW_PREFIX_value=
ac_cv_env_STOW_ROOT_set=
ac_cv_env_STOW_ROOT_value=
ac_cv_env_build_alias_set=set
ac_cv_env_build_alias_value=x86_64-unknown-linux-gnu
ac_cv_env_host_alias_set=set
ac_cv_env_host_alias_value=riscv64-none-elf
ac_cv_env_target_alias_set=
ac_cv_env_target_alias_value=
ac_cv_host=riscv64-none-elf
ac_cv_prog_CC=riscv64-none-elf-gcc
lt_cv_deplibs_check_method=pass_all

## ----------------- ##
## Output variables. ##
## ----------------- ##

AR='riscv64-none-elf-ar'
BBL_ENABLE_LOGO=''
BBL_LOGO_FILE=''
BBL_PAYLOAD=''
CC='riscv64-none-elf-gcc'
CFLAGS=''
CPPFLAGS=''
CUSTOM_DTS=''
CXX='riscv64-none-elf-g++'
CXXFLAGS=''
DEFS=''
ECHO_C=''
ECHO_N='-n'
ECHO_T=''
EGREP=''
EXEEXT=''
GREP=''
INSTALL_DATA=''
INSTALL_PROGRAM=''
INSTALL_SCRIPT=''
LDFLAGS=''
LIBOBJS=''
LIBS=''
LTLIBOBJS=''
MEM_START=''
OBJCOPY='riscv64-none-elf-objcopy'
OBJEXT=''
PACKAGE_BUGREPORT='Andrew Waterman'
PACKAGE_NAME='RISC-V Proxy Kernel'
PACKAGE_STRING='RISC-V Proxy Kernel 1.0.0'
PACKAGE_TARNAME='riscv-pk'
PACKAGE_URL=''
PACKAGE_VERSION='1.0.0'
PATH_SEPARATOR=':'
RANLIB='riscv64-none-elf-ranlib'
READELF='riscv64-none-elf-readelf'
RISCV=''
SHELL='/nix/store/cl2gkgnh26mmpka81pc2g5bzjfrili92-bash-5.3p3/bin/bash'
STOW_PREFIX=''
STOW_ROOT=''
WITH_ABI=''
WITH_ARCH=''
ac_ct_CC=''
ac_ct_CXX=''
bindir='${exec_prefix}/bin'
build='x86_64-unknown-linux-gnu'
build_alias='x86_64-unknown-linux-gnu'
build_cpu='x86_64'
build_os='linux-gnu'
build_vendor='unknown'
datadir='${datarootdir}'
datarootdir='${prefix}/share'
docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
dvidir='${docdir}'
enable_stow=''
exec_prefix='NONE'
host='riscv64-none-elf'
host_alias='riscv64-none-elf'
host_cpu='riscv64'
host_os='elf'
host_vendor='none'
htmldir='${docdir}'
includedir='${prefix}/include'
infodir='${datarootdir}/info'
libdir='${exec_prefix}/lib'
libexecdir='${exec_prefix}/libexec'
localedir='${datarootdir}/locale'
localstatedir='${prefix}/var'
mandir='${datarootdir}/man'
oldincludedir='/usr/include'
pdfdir='${docdir}'
prefix='/nix/store/21cgj30hlk5ys6ax0vzfj64fpy5nrkqp-riscv-pk-riscv64-none-elf-1.0.0-unstable-2024-10-09'
program_transform_name='s,x,x,'
psdir='${docdir}'
runstatedir='${localstatedir}/run'
sbindir='${exec_prefix}/sbin'
sharedstatedir='${prefix}/com'
stow=''
subprojects=''
subprojects_enabled=''
sysconfdir='${prefix}/etc'
target_alias=''

## ----------- ##
## confdefs.h. ##
## ----------- ##

/* confdefs.h */
#define PACKAGE_NAME "RISC-V Proxy Kernel"
#define PACKAGE_TARNAME "riscv-pk"
#define PACKAGE_VERSION "1.0.0"
#define PACKAGE_STRING "RISC-V Proxy Kernel 1.0.0"
#define PACKAGE_BUGREPORT "Andrew Waterman"
#define PACKAGE_URL ""

configure: exit 77
```

### Additional context

Related to #450146 

### System metadata

fetching path input 'path:/nix/store/lsdw9m87nam4syiycx607yjf1b8v9zir-source'
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.49, NixOS, 25.11 (Xantusia), 25.11.20250928.647e5c1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /root/.config/nix/nix.conf:/nix/store/p6vh51114npi9s658cigyyysfcy13g62-stylix-kde-config${XDG_CONFIG_DIRS/nix/nix.conf:+/nix/nix.conf:$XDG_CONFIG_DIRS}/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/zxl9ivn2v7xcrzlf4j036fl6av34zwv9-lix-2.93.3/share`
 - nixpkgs: `/nix/store/lsdw9m87nam4syiycx607yjf1b8v9zir-source`

### Notify maintainers

@shlevy 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/compilers/gcc/default.nix
+++ b/pkgs/development/compilers/gcc/default.nix
@@ -19,7 +19,7 @@
   cargo,
   staticCompiler ? false,
   enableShared ? stdenv.targetPlatform.hasSharedLibraries,
-  enableDefaultPie ? true,
+  enableDefaultPie ? stdenv.targetPlatform.hasSharedLibraries,
   enableLTO ? stdenv.hostPlatform.hasSharedLibraries,
   texinfo ? null,
   perl ? null, # optional, for texi2pod (then pod2man)

